src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodUtils.java

```diff
@@ -57,18 +65,85 @@ public static List<ContainerStatus> getWaitingContainers(Pod pod) {
         return getContainers(pod, CONTAINER_IS_WAITING);
     }
 
-    public static List<ContainerStatus> getContainerStatus(Pod pod) {
+    /**
+     * Get all container statuses (does not include ephemeral or init containers).
+     * @param pod pod to get container statuses for
+     * @return list of statuses, possibly empty, never null
+     */
+    @NonNull
+    public static List<ContainerStatus> getContainerStatus(@NonNull Pod pod) {
         PodStatus podStatus = pod.getStatus();
         if (podStatus == null) {
             return Collections.emptyList();
         }
         return podStatus.getContainerStatuses();
     }
 
-    public static List<ContainerStatus> getContainers(Pod pod, Predicate<ContainerStatus> predicate) {
+    /**
+     * Get pod container statuses (does not include ephemeral or init containers) that match the given filter.
+     * @param pod pod to get container statuses for
+     * @param predicate container status predicate
+     * @return list of statuses, possibly empty, never null
+     */
+    public static List<ContainerStatus> getContainers(@NonNull Pod pod, Predicate<ContainerStatus> predicate) {
         return getContainerStatus(pod).stream().filter(predicate).collect(Collectors.toList());
     }
 
+    /**
+     * Lookup container status (either main container or ephemeral container).
+     * @param pod pod resource to inspect
+     * @param containerName container to locate
+     * @return container status if found
+     */
+    public static Optional<ContainerStatus> getContainerStatus(Pod pod, String containerName) {
```

Comment:
This is not called from anywhere in this plugin. If it's required for the extension plugin, move it there.

Reply:
It will be called in a future PR to that improves the KuberenetesCloud log handling. I can move it to that PR but it was easier to cherry pick it here since I add unit tests for PodUtils which has none before.

https://github.com/cronik/kubernetes-plugin/blob/f4c309255237b3db9fde9084ec126e293cbea984/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java#L167

Reply:
I agree with Antonio, this will be easier for later forensics (if any)